### DISCUSSION TITLE
Subscription not found issue with GenerateResourcesAndImage

### DISCUSSION THREAD

**Discussion by tedm13** [community/community#160670] [Actions]
### Description

I've recently had troubles running the scripts specifically the GenerateResourcesAndImage, the result is with an error saying it is unable to find the Subscription however it will create the resource group but it will not create the storage account to hold the resources. I've had this working in a different subscription but not in a subscription I just created to isolate all the build images. The new subscription has the proper IAM roles and I've tried creating a app registration thinking there was an access issue but it didn't work.

I'm also running this script locally in windows/powershell and have had it working but when switching the target subscription I receive errors.

Below is the command I run.

GenerateResourcesAndImage -SubscriptionId {SubscriptionID} -ResourceGroupName "rg-st-ubuntu-eastus" -ImageGenerationRepositoryRoot "$pwd" -ImageType Ubuntu2004 -AzureLocation "East US" -AzureClientId {AzureClientId} -AzureClientSecret {AzureClientSecret} -AzureTenantId {AzureTenantId }

This is the output that I get

Name               : DevOps Pipelines {IDs}
Account            : {SubsAccount}
Environment        : AzureCloud
Subscription       : {SubscriptionID}
Tenant             : {TenantID}
TokenCache         :
VersionProfile     :
ExtendedProperties : {}

ResourceGroupName : rg-st-ubuntu-eastus
Location          : eastus
ProvisioningState : Succeeded
Tags              :
TagsTable         :
ResourceId        : /subscriptions/{SubscriptionID}/resourceGroups/rg-st-ubuntu-eastus
ManagedBy         :

Delete Resource Group
The resource group you specified already exists. Do you want to clean it up?
[Y] Yes  [N] No  [S] Stop  [?] Help (default is "Y"): N
GenerateResourcesAndImage : Subscription {SubscriptionID} was not found.
At line:1 char:1
+ GenerateResourcesAndImage -SubscriptionId {SubscriptionID} ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,GenerateResourcesAndImage

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions

### Virtual environments affected

- [ ] Ubuntu 18.04
- [X] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

I'm using the latest from master

### Is it regression?

No

### Expected behavior

To be able to generate the resources and images.

### Actual behavior

Erroring out right after the resource group is created.

### Repro steps

1. Import the helper to GenerateResourcesAndImages
2. Run the GenerateResourcesAndImages script
3. It will error out

**Comment by vamshisiram** [comment]
@mikhailkoliada any thoughts into this please? 

i am getting the same exact issue as @tedm13 .

**Comment by vamshisiram** [comment]
Yeah, I don’t understand why your issue and mine were immediately converted to a discussion and no response.

I cannot use the one that works because I have an issue with that subscription, we have defender for cloud enable and the script for image generation fails in the one that used to work.

So, I need to try this on a new subscription.

I hope someone responds to this. Sad part is that we cannot open a Microsoft case as this is a community driven one.  If we can open a ms case I can, can we open one?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by pebroz** [reply]
The same story here. Just got a new subscription as the one with defender does not work. The script creates the resource group, but then fails with GenerateResourcesAndImage: Subscription not found.

**Reply by pkit** [reply]
@pebroz storage is not registered by default on new accounts.
See https://github.com/actions/runner-images/discussions/6163#discussioncomment-4627264

**Comment by sshakhtour** [comment]
I faced similar issue and I resolved issue by registering resource provider of azure storage account 

az provider register –namespace Microsoft.Storage
az provider show –namespace Microsoft.Storage

**Reply by wpyoga** [reply]
I solved my issue with this method too. The only difference is that I did it on the Azure portal as outlined here: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types

I think this needs to go into a FAQ somewhere...

**Comment by mikebeaton** [comment]
I tried various things, but I think it was literally just a matter of waiting < 1 hr. for the new account to be set up.

**Reply by pkit** [reply]
Storage setup will not solve itself in 1 hour

**Reply by mikebeaton** [reply]
I never did the step you and others mentioned [above](https://github.com/actions/runner-images/discussions/6163#discussioncomment-5785950), but it did start working for me after the approx. time mentioned. (If that issue is what is wrong, then I am not saying it will fix itself - although for that matter, maybe it will, idk - just that I didn't manually have to do that step - so either it is initially wrong and it does eventually fix itself, or there are two different possible issues here.)
